File: api-backend/python/change-file-or-folder-owner.py
import requests

# Set your DocSpace portal URL, access token, and IDs
API_HOST = 'https://yourportal.onlyoffice.com'
API_KEY = 'your_api_key'

HEADERS = {
  'Authorization': API_KEY,
  'Content-Type': 'application/json'
}

# Step 1: Define file and/or folder IDs to transfer
file_ids = [12345]  # Replace with your file IDs
folder_ids = [67890]  # Replace with folder IDs if needed
new_owner_id = 'user_uuid_here'  # New owner's UUID

payload = {
  "fileIds": file_ids,
  "folderIds": folder_ids,
  "userId": new_owner_id
}

# Step 2: Send the ownership change request
def change_ownership(payload):
  url = f'{API_HOST}/api/2.0/files/owner'
  response = requests.post(url, headers=HEADERS, json=payload)

  if response.status_code == 200:
    result = response.json().get('response', [])
    print("Ownership successfully updated for the following items:")
    for entry in result:
      title = entry.get('title', 'Unnamed')
      shared = entry.get('access', {}).get('shared', False)
      print(f"- {title} (Shared: {shared})")
  else:
    print(f"Ownership update failed. Status code: {response.status_code}, Message: {response.text}")

# Run the method
if __name__ == '__main__':
  change_ownership(payload)

File: api-backend/nodejs/samples/change-file-or-folder-owner.js
// Set your DocSpace portal URL, access token, and IDs
const BASE_URL = 'https://yourportal.onlyoffice.com';
const API_KEY = 'your_api_key';

const HEADERS = {
  Authorization: API_KEY,
  'Content-Type': 'application/json',
};

// Step 1: Define file and/or folder IDs to transfer
const fileIds = [12345]; // Replace with your file IDs
const folderIds = [67890]; // Replace with folder IDs if needed
const newOwnerId = 'user_uuid_here'; // New owner's UUID

const payload = {
  fileIds: fileIds,
  folderIds: folderIds,
  userId: newOwnerId,
};

// Step 2: Send the ownership change request
async function changeOwnership(payload) {
  const url = `${BASE_URL}/api/2.0/files/owner`;

  try {
    const res = await fetch(url, {
      method: 'POST',
      headers: HEADERS,
      body: JSON.stringify(payload),
    });

    if (res.status !== 200) {
      const text = await res.text();
      console.log(`Ownership update failed. Status code: ${res.status}, Message: ${text}`);
      return;
    }

    const data = await res.json();
    const result = data?.response ?? [];

    console.log('Ownership successfully updated for the following items:');
    result.forEach((entry) => {
      const title = entry?.title ?? 'Unnamed';
      const shared = entry?.access?.shared ?? false;
      console.log(`- ${title} (Shared: ${shared})`);
    });
  } catch (err) {
    console.log(`Ownership update error: ${err}`);
  }
}

// Run the method
changeOwnership(payload);